src/zope/cachedescriptors/property.py

```diff
@@ -51,6 +55,36 @@ def __get__(self, inst, class_):
 
         return value
 
+def CachedProperty(*args):
```

Comment:
Good question, I guess. This fix is based on our own production code, which monkey-patches this module without any problems, but that's hardly conclusive. I imagine it would be difficult to subclass CachedProperty reliably/testably.